### PR TITLE
Rooms router update procedure

### PR DESCRIPTION
**Pull request by Mariphy**
I added a rooms router update procedure. Not sure how to test it without the UI yet. I hope it works. I tried to do more things today, so I created two brunches, but as the first one wasn't merged yet, I manually added the code to the new branch. I hope it won't mess thing up.

### CODE REVIEW

File: src/server/api/routers/rooms.ts [hunk 1]
```diff
@@ -33,6 +34,24 @@ export const roomsRouter = createTRPCRouter({
           id:randomUUID()
         }).returning();
       return res;
-    })
-    
+    }),
+    updateRooms: teacherProcedure
```
Comment (dan2811) [inline src/server/api/routers/rooms.ts:38]: Really good this should work perfectly. Just call this `update` though, that'll keep it inline with the other procedures on this router.

**Comment by dan2811** [comment]
> I added a rooms router update procedure. Not sure how to test it without the UI yet. I hope it works. I tried to do more things today, so I created two brunches, but as the first one wasn't merged yet, I manually added the code to the new branch. I hope it won't mess thing up.

As for the branches. What you've done should be fine, when the first branch gets merged to main, check the diff for this PR by clicking on [files](https://github.com/dan2811/yamaha/pull/20/files) at the top of this PR. You should notice that the changes from the branch that has already been merged disappear from the diff. That's because those changes are in both the main branch and this branch. We can call if this is confusing, it took me a while to get my head round!

An easier workflow for future PRs: 
Assuming we will always merge your PRs in the order that the branches were created, then you can actually branch from a branch that is not merged yet. For Example:

Assuming there are no branches with work in progress open at the moment. I would branch from main (let's call this branch `feature-1`), make my changes, then open a PR (notice that the target branch for this PR is `main`). Then whilst I'm waiting for a review on that PR, I open a new branch, but this time, instead of branching off from `main`, I branch from `feature-1` so now I already have all the changes from `feature-1` and I can add my new changes (let's call this branch `feature-2`). Once I've done my work on `feature-2` I open a PR for it, but this time I set the destination branch to `feature-1` this just means that when I look at the diff for this PR, it is only comparing `feature-2` -> `feature-1` so I won't see the changes from `feature-1`. Then when `feature-1` gets approved and merged into `main`, GitHub will automatically change the target branch of the second PR to main.

But let's say that the PR for the `feature-1` branch hasn't been approved and merged yet, it actually got some comments and I need to make further changes, this is fine. I just make those changes on `feature-1` as normal, then once I've committed and pushed those changes. I can do `git checkout feature-2` and `git merge feature-1` this should update the `feature-2` branch with the latest changes from `feature-1`.

Hopefully that will eventually make sense. I found it really hard to get my head round, but after thinking about it for a while it's very logical and intuitive, it's just remembering the terminology that is the tricky part!

File: src/server/api/routers/rooms.ts (hunk 1 above)
Comment (Mariphy) [reply src/server/api/routers/rooms.ts:38]: Got it!

**Comment by Mariphy** [comment]
Thank you for the detailed explanation! This time I'll wait for the merge of the roomsList branch, then update this one with changes and commit the renaming. 

Next time I will make sure to create the brunch from the issue and then branch the new one from there.

**Comment by Mariphy** [comment]
Hi Dan, can you please merge if everything looks fine?

**Comment by dan2811** [comment]
> Hi Dan, can you please merge if everything looks fine?
Sorry, I keep forgetting that you can't merge. I'm working on fixing the preview deployments at the moment, so hopefully you won't be relying on me merging your PRs for much longer! :D

**Pull request merged**